### ISSUE TITLE
Array as an object when running in browser

### ISSUE THREAD

**Issue by vbarbarosh** [sylvainpolletvillard/ObjectModel#29]
The following code works OK in node, but gives unexpected result when running in browser:
```javascript
var Model = require('objectmodel');

// Default is [1, 0, 0, 1, 0, 0]
var Matrix = Model.Array(Number);
    //.assert(v => v.length == 6);

Matrix.prototype.toString = function () {
    var [a, b, c, d, e, f] = this;
    return `matrix(${a} ${b} ${c} ${d} ${e} ${f})`;
};

var Shape = Model({
    matrix: Matrix,
    path: String
}).defaults({matrix: new Matrix([1, 0, 0, 1, 0, 0])});

var Picture2 = Model({
    matrix: [Matrix],
    shapes: Model.Array(Shape)
}).defaults({matrix: undefined, shapes: []});

var picture = new Picture2({
    shapes: [
        new Shape({
            path: ''
        }),
        new Shape({
            path: ''
        }),
        new Shape({
            path: ''
        }),
        new Shape({
            path: ''
        })
    ]
});

console.log(JSON.parse(JSON.stringify(picture)));
```

Instead of this:

```javascript
{shapes: [..., ..., ..., ...]}
```

I've got this:
```javascript
{shapes: {0: ..., 1: ..., 2: ...}}
```

**Comment by sylvainpolletvillard** [comment]
What browser are you using ? I think it's working as intended but the browser don't know how to convert proxyfied arrays to strings.

On Chrome: 
![chrome_2016-09-03_12-26-46](https://cloud.githubusercontent.com/assets/566536/18224332/e77bf0ee-71d1-11e6-92b8-867769d696d2.png)

On Firefox:
![firefox_2016-09-03_12-27-08](https://cloud.githubusercontent.com/assets/566536/18224331/e7782752-71d1-11e6-8acd-eb831ad1b2f3.png)

**Comment by vbarbarosh** [comment]
Google Chrome Version 52.0.2743.116 (64-bit)

**Comment by sylvainpolletvillard** [comment]
I just tested Chrome v52 on Browserstack and can't reproduce the issue:

![chrome_2016-09-03_13-14-24](https://cloud.githubusercontent.com/assets/566536/18224518/6fde43f0-71d8-11e6-83de-07eac33c07f6.png)

Could you try `console.log(JSON.parse(JSON.stringify(picture)).shapes instanceof Array)` ?

**Comment by vbarbarosh** [comment]
I'll do this on Monday. Sorry for such a delay.

**Comment by sylvainpolletvillard** [comment]
No problem, have a nice week-end :)

**Comment by vbarbarosh** [comment]
[2.3.0](https://jsbin.com/fucucehaka/edit?js,console)
```
TypeError: expecting matrix to be Array of Number, got Object matrix(1 0 0 1 0 0)
```

[2.1.0](https://jsbin.com/coluvomiza/edit?js,console)
```
[object Object] {
  shapes: [object Object] {
    0: [object Object] { ... },
    1: [object Object] { ... },
    2: [object Object] { ... },
    3: [object Object] { ... }
  }
}
true
```
Even though ``instanceof Array`` returns ``true``, this doesn't help me. I use Vue.js and it seems not to recognize ``shapes`` as array. Because of this, when I update ``shapes`` I see no changes in the UI.

Currently I use ``ObjectModel`` for creation only:
```javascript
var project = JSON.parse(JSON.stringify({
    picture: {
        matrix: [1, 0, 0, 1, 0, 0],
        shapes: [
            new ShapeModel({
                ...
            }),
            new ShapeModel({
                ...
            }),
            new ShapeModel({
                ....
            }),
            new ShapeModel({
                ....
            })
        ]
    },
    texts: []
}));

```

BTW I use latest Chrome on Debian 8.5.

**Comment by sylvainpolletvillard** [comment]
Kudos for using Vue.js :+1: This is an excellent framework and Evan is using very similar techniques to ObjectModel for its reactivity system (getters/setters and dependency tracking).

The JSBin will help a lot, I managed to reproduce the issue yet I don't understand why the same code works perfectly in other contexts. Working on it

**Comment by sylvainpolletvillard** [comment]
Got it ! It came from Array.isArray test which does not like proxyfied arrays. Fixed in v2.3.1
